ee/clickhouse/queries/breakdown_props.py

```diff
@@ -7,104 +7,77 @@
-    element_params = {
-        "limit": limit,
-        "team_id": team_id,
-        "offset": filter.offset,
-        **params,
-    }
-
-    top_elements_array = sync_execute(query, element_params)[0][0]
-
-    return top_elements_array
-
-
-def get_breakdown_person_prop_values(
-    filter: Filter, entity: Entity, aggregate_operation: str, team_id: int, limit: int = 25, extra_params={}
+def get_breakdown_prop_values(
+    filter: Filter,
+    entity: Entity,
+    aggregate_operation: str,
+    team_id: int,
+    limit: int = 25,
+    extra_params={},
+    column_optimizer: Optional[ColumnOptimizer] = None,
 ):
-    _event_table_name = "e"
+    "Returns the top N breakdown prop values for event/person breakdown"
+
     parsed_date_from, parsed_date_to, _ = parse_timestamps(filter=filter, team_id=team_id)
     prop_filters, prop_filter_params = parse_prop_clauses(
         filter.properties,
         team_id,
-        table_name=_event_table_name,
+        table_name="e",
+        prepend="e_brkdwn",
         filter_test_accounts=filter.filter_test_accounts,
         person_properties_column="person_props",
         allow_denormalized_props=True,
     )
 
     entity_params, entity_format_params = get_entity_filtering_params(
-        entity, team_id, table_name=_event_table_name, with_prop_filters=True, person_properties_column="person_props"
+        entity, team_id, with_prop_filters=True, person_properties_column="person_props", table_name="e"
     )
 
-    value_expression, _ = get_property_string_expr("person", cast(str, filter.breakdown), "%(key)s", "person_props")
-
-    elements_query = TOP_PERSON_PROPS_ARRAY_OF_KEY_SQL.format(
-        value_expression=value_expression,
-        parsed_date_from=parsed_date_from,
-        parsed_date_to=parsed_date_to,
-        person_query=ClickhousePersonQuery(filter, team_id).get_query(),
-        prop_filters=prop_filters,
-        aggregate_operation=aggregate_operation,
-        GET_TEAM_PERSON_DISTINCT_IDS=GET_TEAM_PERSON_DISTINCT_IDS,
-        **entity_format_params,
-    )
-    top_elements_array = _get_top_elements(
-        filter=filter,
-        team_id=team_id,
-        query=elements_query,
-        params={**prop_filter_params, **entity_params, **extra_params, "key": filter.breakdown},
-        limit=limit,
-    )
-
-    return top_elements_array
-
-
-def get_breakdown_event_prop_values(
-    filter: Filter, entity: Entity, aggregate_operation: str, team_id: int, limit: int = 25, extra_params={}
-):
-    parsed_date_from, parsed_date_to, _ = parse_timestamps(filter=filter, team_id=team_id)
-    prop_filters, prop_filter_params = parse_prop_clauses(
-        filter.properties,
-        team_id,
-        table_name="e",
-        prepend="e_brkdwn",
-        filter_test_accounts=filter.filter_test_accounts,
-    )
-
-    entity_params, entity_format_params = get_entity_filtering_params(entity, team_id, with_prop_filters=True)
+    if filter.breakdown_type == "person":
```

Comment:
Do we need to consider `"cohorts"` here as well as a `breakdown_type`. I'm not sure if it's relevant or what the other values of `breakdown_type` can be.

Reply:
The typing here is a little unclear but the "cohort" value is handled in ee/clickhouse/queries/trends/breakdown.py line 84. This function specifically deals with person/event properties which are queried/formatted differently than the cohort ids on a breakdown